modules/core/src/main/java/org/apache/ignite/internal/processors/metastorage/persistence/DistributedMetaStorageImpl.java

```diff
@@ -703,18 +704,33 @@ else if (remoteVer.id < locVer.id) {
 
                 if (newRemoteVer.equals(locVer))
                     errorMsg = null;
-                else
-                    errorMsg = "Joining node has conflicting distributed metastorage data.";
+                else {
+                    DistributedMetaStorageHistoryItem[] locHist = localHistoryArray(remoteVer.id + 1, locVer.id);
+
+                    errorMsg = S.toString(
```

Comment:
I'm curious, why not just use a formatted string?

Reply:
Tradition, we use this method for logs very often. I wonder why we use formatting if it exists, actually